src/jumpstart/saga.ts

```diff
@@ -0,0 +1,199 @@
+import Logger from 'src/utils/Logger'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { publicClient } from 'src/viem'
+import { networkIdToNetwork } from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { all, call, fork, put, select } from 'typed-redux-saga'
+import { Hash, TransactionReceipt, parseAbi, parseEventLogs } from 'viem'
+
+const TAG = 'WalletJumpstart'
+
+export function* jumpstartClaim(privateKey: string, networkId: NetworkId) {
+  try {
+    yield* put(jumpstartClaimStarted())
+
+    const walletAddress = yield* select(walletAddressSelector)
+    if (!walletAddress) {
+      throw new Error('No wallet address found in store. This should never happen.')
+    }
+
+    const contractAddress = getDynamicConfigParams(
+      DynamicConfigs[StatsigDynamicConfigs.WALLET_JUMPSTART_CONFIG]
+    ).jumpstartContracts?.[networkId]?.contractAddress
+
+    if (!contractAddress) {
+      throw new Error(`Contract address for ${networkId} is not provided in dynamic config`)
+    }
+
+    const transactionHashes = yield* call(
+      jumpstartLinkHandler,
+      networkId,
+      contractAddress,
+      privateKey,
+      walletAddress
+    )
+
+    yield* fork(dispatchPendingTransactions, networkId, transactionHashes)
+
+    ValoraAnalytics.track(JumpstartEvents.jumpstart_claim_succeeded)
+
+    yield* put(jumpstartClaimSucceeded())
+  } catch (error) {
+    Logger.error(TAG, 'Error handling jumpstart link', error)
+    ValoraAnalytics.track(JumpstartEvents.jumpstart_claim_failed)
+    yield* put(jumpstartClaimFailed())
+  }
+}
+
+export function* dispatchPendingTransactions(networkId: NetworkId, transactionHashes: Hash[]) {
+  try {
+    const network = networkIdToNetwork[networkId]
+    const transactionReceipts: TransactionReceipt[] = yield* all(
+      transactionHashes.map((hash) =>
+        call([publicClient[network], 'getTransactionReceipt'], { hash })
+      )
+    )
+
+    yield* fork(dispatchPendingERC20Transactions, networkId, transactionReceipts)
+    yield* fork(dispatchPendingERC721Transactions, networkId, transactionReceipts)
+  } catch (error) {
+    Logger.error(TAG, 'Error dispatching pending transactions', error)
+  }
+}
+
+export function* dispatchPendingERC20Transactions(
+  networkId: NetworkId,
+  transactionReceipts: TransactionReceipt[]
+) {
+  const tokensById = yield* select((state) => tokensByIdSelector(state, [networkId]))
+
+  for (const { transactionHash, logs } of transactionReceipts) {
+    const parsedLogs = parseEventLogs({
+      abi: walletJumpstart.abi,
+      eventName: ['ERC20Claimed'],
+      logs,
+    })
+
+    for (const {
+      address,
+      args: { token: tokenAddress, amount },
+    } of parsedLogs) {
+      const tokenId = getTokenId(networkId, tokenAddress)
+
+      const token = tokensById[tokenId]
+      if (!token) {
+        Logger.error(TAG, 'Claimed unknown tokenId', tokenId)
```

Comment:
```suggestion
        Logger.warn(TAG, 'Claimed unknown tokenId', tokenId)
```
seems that we wouldn't want to investigate this right?